src/widgets/stats/stats.tsx

```diff
@@ -0,0 +1,165 @@
+/** @jsx h */
+
+import { h, render } from 'preact';
+import cx from 'classnames';
+import Stats from '../../components/Stats/Stats';
+import connectStats, {
+  StatsConnectorParams,
+  StatsRendererOptions,
+} from '../../connectors/stats/connectStats';
+import {
+  prepareTemplateProps,
+  getContainerNode,
+  createDocumentationMessageGenerator,
+} from '../../lib/utils';
+import { component } from '../../lib/suit';
+import { Renderer, Template, WidgetFactory } from '../../types';
+
+const withUsage = createDocumentationMessageGenerator({ name: 'stats' });
+const suit = component('Stats');
+
+export type StatsCSSClasses = {
+  /**
+   * CSS class to add to the root element.
+   */
+  root: string | string[];
+
+  /**
+   * CSS class to add to the text span element.
+   */
+  text: string | string[];
+};
+
+export type StatsTemplates = {
+  /**
+   * Text template, provided with `hasManyResults`, `hasNoResults`, `hasOneResult`, `hitsPerPage`, `nbHits`, `nbSortedHits`, `nbPages`, `page`, `processingTimeMS`, `query`.
+   */
+  text: Template;
```

Comment:
```suggestion
  text: Template<{hasManuResults.....>;
```

Reply:
Fixed in https://github.com/algolia/instantsearch.js/pull/4681/commits/d9217028858b5fdd2bce87a72b38438d373d0f3f